test/relations.test.js

```diff
@@ -1947,30 +2671,68 @@ describe('relations', function () {
           supplierId = supplier.id;
           should.not.exist(e);
           should.exist(supplier);
-          supplier.account.create({accountNo: 'a01'}, function (err, account) {
+          supplier.account.create({accountNo: 'a01', block: false}, function (err, account) {
             supplier.account(function (e, act) {
               accountId = act.id;
               should.not.exist(e);
               should.exist(act);
               act.should.be.an.instanceOf(Account);
+              act.block.should.be.false;
               supplier.account().id.should.equal(act.id);
               act.supplierName.should.equal(supplier.name);
               done();
             });
           });
-        });
+        });
+      });
+    });
+
+    it('should find record that match scope', function (done) {
+      Account.updateAll({ block: true }, function (err) {
+        Supplier.findById(supplierId, function (err, supplier) {
+          supplier.account(function (err, account) {
+            should.not.exists(account);
+            done();
+          });
+        });
+      });
+    });
+
+    it('can be used to query data with promises', function (done) {
+      db.automigrate(function () {
+        Supplier.create({name: 'Supplier 1'})
+        .then(function (supplier) {
+          supplierId = supplier.id;
+          should.exist(supplier);
+          return supplier.account.create({accountNo: 'a01', block: false})
+          .then(function (account) {
+            return supplier.account.get()
+          })
+          .then(function (act) {
+            accountId = act.id;
+            should.exist(act);
+            act.should.be.an.instanceOf(Account);
+            act.block.should.be.false;
```

Comment:
I think the diff was just confusing that in this section because I grouped the promise tests together rather than interleaving them.

Some of the tests in this section rely on data created in the previous test(s), so I kept the flow between dependent tests rather than rewrite them completely.